src/Symfony/Bundle/TwigBundle/DependencyInjection/Compiler/ExtensionPass.php

```diff
@@ -72,15 +76,18 @@ public function process(ContainerBuilder $container)
             $container->getDefinition('twig.extension.assets')->addTag('twig.extension');
         }
 
-        if (class_exists('Symfony\Component\Yaml\Parser')) {
+        $container->addResource(new ClassExistenceResource(YamlParser::class));
+        if (class_exists(YamlParser::class)) {
```

Comment:
Related to [this comment](https://github.com/symfony/symfony/commit/79efb4c8414ea459d1ad678ed79ba469e3597d51#commitcomment-19258739), just asking if could we replace this:

```php
if (class_exists(YamlParser::class)) {
    $container->getDefinition('twig.extension.yaml')->addTag('twig.extension');
}
```

by this?

```php
if (!class_exists(YamlParser::class)) {
    $container->removeDefinition('twig.extension.yaml');
}
```